src/Workspaces/Core/Portable/Rename/ConflictEngine/ConflictResolver.Session.cs

```diff
@@ -842,6 +845,14 @@ private async Task<Solution> AnnotateAndRename_WorkerAsync(
                     }
                     else
                     {
+                        // In a source generated document, we have to ensure we've realized the "old" tree in the modified solution of WithDocumentSyntaxRoot
+                        // won't work. Performing a rename in a source generated document is opt-in, so we can assume that we only hit this condition in
+                        // scenarios that wanted it.
+                        if (documentId.IsSourceGenerated)
+                        {
+                            _ = await partiallyRenamedSolution.GetRequiredDocumentAsync(documentId, includeSourceGenerated: true, _cancellationToken).ConfigureAwait(false);
```

Comment:
When I did the changes to allow `WithDocumentXXX`  to work on source generated documents, I made it only work for already generated documents because Razor could guarantee that, and it seemed less risky for any other scenario. This is the downside though, where this (and the other spot it appears) unfortunately has to force the generated document to be created, just to change it again. For _now_ at least I think it's better to do this, than to allow the `WithDocumentXXX` methods to work on unrealised source generated documents. As per the comment in the file, this should only happen when Razor explicitly opts in to rename in source generated documents.